buildkite/resource_team_test.go

```diff
@@ -0,0 +1,193 @@
+					resource.TestCheckResourceAttr("buildkite_team.test", "name", "wombats"),
+					resource.TestCheckResourceAttr("buildkite_team.test", "name", "wombats"),
+				),
+			},
+			{
+				Config: testAccTeamConfigSecret("wombats"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the team exists in the buildkite API
+					testAccCheckTeamExists("buildkite_team.test", &resourceTeam),
+					// Confirm the team has the updated values in Buildkite's system
+					testAccCheckTeamRemoteValues(&resourceTeam, "wombats"),
+					// Confirm the team has the updated values in terraform state
+					resource.TestCheckResourceAttr("buildkite_team.test", "name", "wombats"),
+					resource.TestCheckResourceAttr("buildkite_team.test", "description", "a secret team of wombats"),
+					resource.TestCheckResourceAttr("buildkite_team.test", "privacy", "SECRET"),
+				),
+			},
+		},
+	})
+}
+
+// Confirm that this resource can be imported
+func TestAccTeam_import(t *testing.T) {
+	var resourceTeam TeamNode
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckExampleResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTeamConfigBasic("important"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					// Confirm the team exists in the buildkite API
+					testAccCheckTeamExists("buildkite_team.test", &resourceTeam),
+					// Quick check to confirm the local state is correct before we re-import it
+					resource.TestCheckResourceAttr("buildkite_team.test", "name", "important"),
+				),
+			},
+			{
+				// re-import the resource (using the graphql token of the existing resource) and confirm they match
+				ResourceName:      "buildkite_team.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckTeamExists(resourceName string, resourceTeam *TeamNode) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		resourceState, ok := s.RootModule().Resources[resourceName]
+
+		if !ok {
+			return fmt.Errorf("Not found in state: %s", resourceName)
+		}
+
+		if resourceState.Primary.ID == "" {
+			return fmt.Errorf("No ID is set in state")
+		}
+
+		provider := testAccProvider.Meta().(*Client)
+		var query struct {
+			Node struct {
+				Team TeamNode `graphql:"... on Team"`
+			} `graphql:"node(id: $id)"`
+		}
+
+		vars := map[string]interface{}{
+			"id": resourceState.Primary.ID,
+		}
+
+		err := provider.graphql.Query(context.Background(), &query, vars)
+		if err != nil {
+			return fmt.Errorf("Error fetching team from graphql API: %v", err)
+		}
+
+		if string(query.Node.Team.ID) == "" {
+			return fmt.Errorf("No team found with graphql id: %s", resourceState.Primary.ID)
+		}
+
+		if string(query.Node.Team.Name) != resourceState.Primary.Attributes["name"] {
+			return fmt.Errorf("Team name in state doesn't match remote name")
+		}
```

Comment:
In our other resource tests I think we only checked computer properties in this function, and I also note that name is checked in `testAccCheckTeamRemoteValues`. Do we definitely want to check it here too?

Reply:
Probably excessive, I'll remove it!